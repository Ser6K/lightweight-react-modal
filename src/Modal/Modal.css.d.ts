declare namespace ModalCssNamespace {
  export interface IModalCss {
    modal: string;
    wrapper: string;
  }
}

declare const ModalCssModule: ModalCssNamespace.IModalCss & {
  /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
  locals: ModalCssNamespace.IModalCss;
};

export = ModalCssModule;
