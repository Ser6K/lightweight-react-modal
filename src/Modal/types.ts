import { ClassNameType, ReactNodeType } from 'src/types'

export interface ModalTypes {
  onClose?: () => void
  fluid?: boolean
  closable?: boolean
  maxHeight?: number
  minHeight?: number
  maxWidth?: number
  minWidth?: number
  children: ReactNodeType
  customClassNames?: {
    wrapper: ClassNameType
    modal: ClassNameType
    closeBtn: ClassNameType
    overlay: ClassNameType
  }
  closeButtonIcon?: ReactNodeType
  [propName: string]: any
}
