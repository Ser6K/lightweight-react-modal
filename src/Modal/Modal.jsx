import React, {
    useState,
    useCallback,
    useEffect,
    useRef,
} from 'react';
import { createPortal } from 'react-dom';
import classNames from 'classnames';
import { createUseStyles } from 'react-jss';
import PropTypes from 'prop-types';
import {
    getHeader,
    getFooter,
    getContent,
    handleEscPress,
} from '../utils/utils';
import { addModal, removeModal } from '../utils/register';
import CloseButton from './components/CloseButton/CloseButton';
import Overlay from './components/Overlay/Overlay';
import styles from './Modal.styles';

const Modal = ({
    onClose,
    fluid,
    closable,
    maxHeight,
    minHeight,
    maxWidth,
    minWidth,
    children,
    customClassNames,
}) => {
    const header = useCallback(getHeader(children), [children]);
    const footer = useCallback(getFooter(children), [children]);
    const content = useCallback(getContent(children), [children]);
    const modalRef = useRef();

    const classes = createUseStyles(styles)({
        fluid, maxHeight, minHeight, maxWidth, minWidth,
    });

    useEffect(() => {
        addModal(modalRef);
    }, [modalRef]);

    const escPressed = handleEscPress(closable, modalRef);
    const closeModal = useCallback(() => {
        if (!closable || !onClose) {
            return;
        }
        removeModal(modalRef);
        onClose();
    }, [closable, onClose, modalRef]);

    useEffect(() => {
        if (escPressed) {
            closeModal();
        }
    }, [escPressed]);

    return createPortal(
        <div className={classNames(classes.wrapper, customClassNames.wrapper)}>
            <div
                ref={modalRef}
                className={classNames(classes.modal, customClassNames.modal)}
            >
                {closable && (
                    <CloseButton
                        className={customClassNames.closeButton}
                        onClick={closeModal}
                    />
                )}
                {header && header}
                {content && content}
                {footer && footer}
            </div>
            <Overlay
                onClick={closeModal}
                className={customClassNames.overlay}
            />
        </div>, document.body,
    );
};

Modal.propTypes = {
    onClose: PropTypes.func,
    fluid: PropTypes.bool,
    closable: PropTypes.bool,
    maxHeight: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
    ]),
    minHeight: PropTypes.number,
    maxWidth: PropTypes.number,
    minWidth: PropTypes.number,
    children: PropTypes.node.isRequired,
    customClassNames: PropTypes.shape({
        wrapper: PropTypes.string,
        modal: PropTypes.string,
        closeBtn: PropTypes.string,
        overlay: PropTypes.string,
    }),
};

Modal.defaultProps = {
    onClose: null,
    fluid: false,
    closable: true,
    maxHeight: 500,
    minHeight: 100,
    maxWidth: 500,
    minWidth: 200,
    customClassNames: {
        wrapper: null,
        modal: null,
        closeButton: null,
        overlay: null,
    },
};

Modal.displayName = 'Modal';
export default React.memo(Modal);
