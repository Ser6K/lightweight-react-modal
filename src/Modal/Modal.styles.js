import styled from 'styled-components';

export const Wrapper = styled.div`
    position: fixed;
    z-index: 99998;
    top: 0;
    bottom: 0;
    right: 0;
    left: 0;
    display: flex;
    align-items: center;
    justify-content: center;
`;

export const ModalWrapper = styled.div`
    position: relative;
    z-index: 99999;
    background: white;
    box-sizing: border-box;
    border: 1px solid grey;
    border-radius: 3px;
    height: 90%;
    width: 90%;
    max-width: ${props => !props.fluid ? `${props.maxWidth}px` : ''};
    min-width: ${props => !props.fluid ? `${props.minWidth}px` : ''};
    max-height: ${props => !props.fluid ? `${props.maxHeight}px` : ''};
    min-height: ${props => !props.fluid ? `${props.minHeight}px` : ''};
    display: flex;
    flex-direction: column;
    flex-wrap: nowrap;
`;
