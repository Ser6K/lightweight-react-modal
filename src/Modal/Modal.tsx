import React, { useCallback, useEffect, useRef } from 'react'
import { createPortal } from 'react-dom'
import { getHeader, getFooter, getContent, handleEscPress, classNames, addModal, removeModal } from 'src/utils'
import CloseButton from './components/CloseButton/CloseButton'
import Overlay from './components/Overlay/Overlay'
import styles from './Modal.css'

interface ModalProps {
  onClose?: () => void
  fluid?: boolean
  closable?: boolean
  maxHeight?: number
  minHeight?: number
  maxWidth?: number
  minWidth?: number
  children: React.ReactNode
  customClassNames?: {
    wrapper: string | null
    modal: string | null
    closeBtn: string | null
    overlay: string | null
  }
  closeButtonIcon?: React.ReactNode
  [propName: string]: any
}

const Modal: React.FC<ModalProps> = ({
  onClose,
  fluid = false,
  closable = true,
  maxHeight = 500,
  minHeight = 100,
  maxWidth = 500,
  minWidth = 200,
  children = null,
  customClassNames = {
    wrapper: null,
    modal: null,
    closeBtn: null,
    overlay: null,
  },
  closeButtonIcon = null,
  ...props
}) => {
  const header = useCallback(getHeader(children), [children])
  const footer = useCallback(getFooter(children), [children])
  const content = useCallback(getContent(children), [children])
  const modalRef = useRef(null)

  useEffect(() => {
    addModal(modalRef)
  }, [modalRef])

  const escPressed = handleEscPress(closable, modalRef)

  const closeModal = useCallback(() => {
    if (!closable || !onClose) {
      return
    }
    removeModal(modalRef)
    onClose()
  }, [closable, onClose, modalRef])

  useEffect(() => {
    if (escPressed) {
      closeModal()
    }
  }, [escPressed])

  return createPortal(
    <div {...props} className={classNames(styles.wrapper, customClassNames.wrapper)}>
      <div
        ref={modalRef}
        className={classNames(styles.modal, customClassNames.modal)}
        style={{
          maxWidth: `${!fluid ? `${maxWidth}px` : ''}`,
          minWidth: `${!fluid ? `${minWidth}px` : ''}`,
          maxHeight: `${!fluid ? `${maxHeight}px` : ''}`,
          minHeight: `${!fluid ? `${minHeight}px` : ''}`,
        }}
      >
        {closable && (
          <CloseButton onClick={closeModal} icon={closeButtonIcon} {...(customClassNames.closeBtn != null ? { className: customClassNames.closeBtn } : {})} />
        )}
        {header && header}
        {content && content}
        {footer && footer}
      </div>
      <Overlay onClick={closeModal} className={customClassNames.overlay} />
    </div>,
    document.body
  )
}

export default React.memo(Modal)
