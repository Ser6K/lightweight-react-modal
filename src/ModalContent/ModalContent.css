.content {
  flex-grow: 1;
  overflow: auto;
  -webkit-overflow-scrolling: touch;
  height: auto;
  margin: 20px 15px;
}
