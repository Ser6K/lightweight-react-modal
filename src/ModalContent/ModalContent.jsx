import React from 'react';
import PropTypes from 'prop-types';
import { ContentDisplayName } from 'utils';
import { Content } from './ModalContent.styles';

const ModalContent = React.memo(({ children, ...props }) => (
    <Content {...props}>
        {children}
    </Content>
));

ModalContent.propTypes = {
    children: PropTypes.node.isRequired,
};

ModalContent.displayName = ContentDisplayName;
export default ModalContent;
