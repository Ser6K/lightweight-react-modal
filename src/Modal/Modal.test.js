import React, { useState } from 'react';
import ReactDom from 'react-dom';
import { shallow, mount } from 'enzyme';
import Modal from './Modal';
import * as ModalRegister from '../utils/register';
import ModalHeader from '../ModalHeader/ModalHeader';
import ModalFooter from '../ModalFooter/ModalFooter';

describe('Modal tests', () => {
    let WrappedModal;

    beforeEach(() => {
        WrappedModal = () => {
            const [open, setOpen] = useState(true);

            return open && (
                <Modal onClose={() => setOpen(false)}>
                    Modal
                </Modal>
            );
        };
    });

    test('it can be mounted', () => {
        const modal = mount(<Modal>Modal</Modal>);
        expect(modal.children()).toHaveLength(1);
    });

    test('it matches snapshot', () => {
        const modal = mount(<Modal><div>dialog</div><div>contetn</div></Modal>);
        expect(modal).toMatchSnapshot();
    });

    test('it can register modal reference on mount', () => {
        const modalRegisterSpy = jest.spyOn(ModalRegister, 'addModal');
        mount(<Modal>Modal</Modal>);

        expect(modalRegisterSpy).toHaveBeenCalledTimes(1);
    });

    test('it mounts with create portal', () => {
        const portalSpy = jest.spyOn(ReactDom, 'createPortal');
        shallow(<Modal>Modal</Modal>);

        expect(portalSpy).toHaveBeenCalledTimes(1);
    });

    test('it can be mounted with header', () => {
        const modal = mount(<Modal><ModalHeader>Header</ModalHeader></Modal>);
        expect(modal.find('ModalHeader').text()).toBe('Header');
    });

    test('it doesn\'t mount header if it\'s empty', () => {
        const modal = mount(<Modal><ModalHeader /></Modal>);
        expect(modal.find('ModalHeader')).toEqual({});
    });

    test('it can be mounted with footer', () => {
        const modal = mount(<Modal><ModalFooter>Footer</ModalFooter></Modal>);
        expect(modal.find('ModalFooter').text()).toBe('Footer');
    });

    test('it doesn\'t mount footer if it\'s empty', () => {
        const modal = mount(<Modal><ModalFooter /></Modal>);
        expect(modal.find('ModalFooter')).toEqual({});
    });

    test('fluid modal matches snapshot', () => {
        const modal = mount(<Modal fluid>Modal</Modal>);
        expect(modal).toMatchSnapshot();
    });

    test('not closable modal matches snapshot', () => {
        const modal = mount(<Modal closable={false}>Modal</Modal>);
        expect(modal).toMatchSnapshot();
    });

    test('it closes with close btn click', () => {
        const modal = mount(<WrappedModal />);
        const closeBtn = modal.find('closeButton');
        closeBtn.simulate('click');
        expect(modal.children()).toHaveLength(0);
    });

    test('it closes with overlay click', () => {
        const modal = mount(<WrappedModal />);
        const modalWrapper = modal.find('div[onClick]');
        modalWrapper.simulate('click');
        expect(modal.children()).toHaveLength(0);
    });

    test('it closes with close btn click', () => {
        const modal = mount(<WrappedModal />);
        const modalRegisterSpy = jest.spyOn(ModalRegister, 'removeModal');
        const closeBtn = modal.find('closeButton');
        closeBtn.simulate('click');
        expect(modalRegisterSpy).toHaveBeenCalledTimes(1);
    });

    test('it can unsubscribe reference on unmount with overlay click', () => {
        const modal = mount(<WrappedModal />);
        const modalRegisterSpy = jest.spyOn(ModalRegister, 'removeModal');
        const modalWrapper = modal.find('div[onClick]');
        modalWrapper.simulate('click');
        expect(modalRegisterSpy).toHaveBeenCalledTimes(1);
    });
});
