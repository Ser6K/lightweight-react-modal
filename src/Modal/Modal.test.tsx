import React from 'react';
import { act } from 'react-dom/test-utils';
import { shallow, mount } from 'enzyme';
import document from 'global/document';
import ReactDom from 'react-dom';
import * as ModalRegister from 'register';
import Modal from './Modal';
import ModalHeader from '../ModalHeader/ModalHeader';
import ModalFooter from '../ModalFooter/ModalFooter';
import ModalContent from '../ModalContent/ModalContent';

describe('Modal snapshots', () => {
    test('default modal matches snapshot', () => {
        const modal = mount(<Modal><div>dialog</div><div>contetn</div></Modal>);
        expect(modal).toMatchSnapshot();
    });

    test('fluid modal matches snapshot', () => {
        const modal = mount(<Modal>Modal</Modal>);
        expect(modal).toMatchSnapshot();
    });

    test('modal matches snapshot when maxHeight/minHeight/maxWidth/minWidth props are provided', () => {
        const modal = mount(
            <Modal
                minHeight={50}
                maxHeight={100}
                minWidth={60}
                maxWidth={120}
            >
                Modal
            </Modal>
        );
        expect(modal).toMatchSnapshot();
    });

    test('modal matches snapshot when custom styles are set', () => {
        const modal = mount(
            <Modal
                customClassNames={{
                    wrapper: 'wrapperClassName',
                    modal: 'modalClassName',
                    closeBtn: 'closeButtonClassName',
                    overlay: 'overlayClassName',
                }}
            >
                Modal
            </Modal>
        );
        expect(modal).toMatchSnapshot();
    });
});

describe('Modal general', () => {
    test('it can be mounted', () => {
        const modal = mount(<Modal>Modal</Modal>);
        expect(modal.children()).toHaveLength(1);
    });

    test('it can register modal reference on mount', () => {
        const modalRegisterSpy = jest.spyOn(ModalRegister, 'addModal');
        mount(<Modal>Modal</Modal>);

        expect(modalRegisterSpy).toHaveBeenCalledTimes(1);
    });

    test('it removes registered modal reference onClose', () => {
        const modal = mount(<Modal onClose={() => {}}>Modal</Modal>);
        const modalUnregisterSpy = jest.spyOn(ModalRegister, 'removeModal');

        modal.find('Overlay').simulate('click');
        expect(modalUnregisterSpy).toHaveBeenCalledTimes(1);
    });

    test('it mounts with create portal', () => {
        const portalSpy = jest.spyOn(ReactDom, 'createPortal');
        shallow(<Modal>Modal</Modal>);

        expect(portalSpy).toHaveBeenCalledTimes(1);
    });

    test('it can be mounted with header, content, footer', () => {
        const modal = mount(
            <Modal>
                <ModalHeader>Header</ModalHeader>
                <ModalContent>Content</ModalContent>
                <ModalFooter>Footer</ModalFooter>
            </Modal>
        );
        expect(modal.find('ModalHeader').text()).toBe('Header');
        expect(modal.find('ModalContent').text()).toBe('Content');
        expect(modal.find('ModalFooter').text()).toBe('Footer');
    });

    test('it does\'t mount header, content, footer if they are empty', () => {
        const modal = mount(
            <Modal>
                <ModalHeader><></></ModalHeader>
                <ModalContent><></></ModalContent>
                <ModalFooter><></></ModalFooter>
            </Modal>
        );
        expect(modal.find('ModalHeader')).toEqual({});
        expect(modal.find('ModalContent')).toEqual({});
        expect(modal.find('ModalFooter')).toEqual({});
    });

    test('onClose button not visible if modal is not closable', () => {
        const modal = mount(<Modal closable={false}>Modal</Modal>);
        const closeButton = modal.find('CloseButton');

        expect(closeButton.exists()).toBeFalsy();
    });
});

describe('Modal props', () => {
    test('onClose fires on overlay click if modal is closable', () => {
        const onClose = jest.fn();
        const modal = mount(<Modal onClose={onClose}>Modal</Modal>);

        modal.find('Overlay').simulate('click');
        expect(onClose).toBeCalledTimes(1);
    });

    test('onClose doesn\'t fire up on overlay click if dialog is not closable', () => {
        const onClose = jest.fn();
        const modal = mount(<Modal closable={false} onClose={onClose}>Modal</Modal>);

        modal.find('Overlay').simulate('click');
        expect(onClose).toBeCalledTimes(0);
    });

    test('onClose fires on close button click', () => {
        const onClose = jest.fn();
        const modal = mount(<Modal onClose={onClose}>Modal</Modal>);

        modal.find('CloseButton').simulate('click');
        expect(onClose).toBeCalledTimes(1);
    });

    describe('Esc key press', () => {
        let events = {};

        beforeEach(() => {
            document.addEventListener = jest.fn((event, cb) => {
                events[event] = cb;
            });
        });

        afterEach(() => {
            events = {};
        });

        test('onClose fires on escape on press', () => {
            const onClose = jest.fn();
            mount(<Modal onClose={onClose}>Modal</Modal>);

            act(() => { events.keydown({ keyCode: 27 }); });
            expect(onClose).toBeCalledTimes(1);
        });

        test('keydown event is not added to document if modal is not closable', () => {
            const onClose = jest.fn();
            mount(<Modal closable={false} onClose={onClose}>Modal</Modal>);

            expect(events).toEqual({});
        });
    });

    test('close button default icon can be changed', () => {
        const onClose = jest.fn();
        const modal = mount(<Modal onClose={onClose} closeButtonIcon="Close">Modal</Modal>);

        const closeButton = modal.find('CloseButton');
        expect(closeButton.text()).toBe('Close');
    });
});