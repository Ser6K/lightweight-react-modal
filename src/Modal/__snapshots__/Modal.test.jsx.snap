// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal snapshots default modal matches snapshot 1`] = `
<Modal>
  <Portal
    containerInfo={
      <body>
        <div
          class="Modalstyles__Wrapper-sc-5bf2l8-0 jIsDqI"
        >
          <div
            class="Modalstyles__ModalWrapper-sc-5bf2l8-1 ffLOAg"
          >
            <button
              class="CloseButtonstyles__Button-sc-1sd1qj0-0 pzudt"
              role="button"
            >
              x
            </button>
          </div>
          <div
            class="Overlaystyles__OverlayWrapper-sc-268q0n-0 giiUMR"
          />
        </div>
      </body>
    }
  >
    <Modalstyles__Wrapper>
      <div
        className="Modalstyles__Wrapper-sc-5bf2l8-0 jIsDqI"
      >
        <Modalstyles__ModalWrapper
          fluid={false}
          maxHeight={500}
          maxWidth={500}
          minHeight={100}
          minWidth={200}
        >
          <div
            className="Modalstyles__ModalWrapper-sc-5bf2l8-1 ffLOAg"
          >
            <CloseButton
              className={null}
              icon={null}
              onClick={[Function]}
            >
              <CloseButtonstyles__Button
                onClick={[Function]}
                role="button"
              >
                <button
                  className="CloseButtonstyles__Button-sc-1sd1qj0-0 pzudt"
                  onClick={[Function]}
                  role="button"
                >
                  x
                </button>
              </CloseButtonstyles__Button>
            </CloseButton>
          </div>
        </Modalstyles__ModalWrapper>
        <Overlay
          className={null}
          onClick={[Function]}
        >
          <Overlaystyles__OverlayWrapper
            onClick={[Function]}
          >
            <div
              className="Overlaystyles__OverlayWrapper-sc-268q0n-0 giiUMR"
              onClick={[Function]}
            />
          </Overlaystyles__OverlayWrapper>
        </Overlay>
      </div>
    </Modalstyles__Wrapper>
  </Portal>
</Modal>
`;

exports[`Modal snapshots fluid modal matches snapshot 1`] = `
<Modal
  fluid={true}
>
  <Portal
    containerInfo={
      <body>
        <div
          class="Modalstyles__Wrapper-sc-5bf2l8-0 jIsDqI"
        >
          <div
            class="Modalstyles__ModalWrapper-sc-5bf2l8-1 ffLOAg"
          >
            <button
              class="CloseButtonstyles__Button-sc-1sd1qj0-0 pzudt"
              role="button"
            >
              x
            </button>
          </div>
          <div
            class="Overlaystyles__OverlayWrapper-sc-268q0n-0 giiUMR"
          />
        </div>
        <div
          class="Modalstyles__Wrapper-sc-5bf2l8-0 jIsDqI"
        >
          <div
            class="Modalstyles__ModalWrapper-sc-5bf2l8-1 caGlFr"
          >
            <button
              class="CloseButtonstyles__Button-sc-1sd1qj0-0 pzudt"
              role="button"
            >
              x
            </button>
          </div>
          <div
            class="Overlaystyles__OverlayWrapper-sc-268q0n-0 giiUMR"
          />
        </div>
      </body>
    }
  >
    <Modalstyles__Wrapper>
      <div
        className="Modalstyles__Wrapper-sc-5bf2l8-0 jIsDqI"
      >
        <Modalstyles__ModalWrapper
          fluid={true}
          maxHeight={500}
          maxWidth={500}
          minHeight={100}
          minWidth={200}
        >
          <div
            className="Modalstyles__ModalWrapper-sc-5bf2l8-1 caGlFr"
          >
            <CloseButton
              className={null}
              icon={null}
              onClick={[Function]}
            >
              <CloseButtonstyles__Button
                onClick={[Function]}
                role="button"
              >
                <button
                  className="CloseButtonstyles__Button-sc-1sd1qj0-0 pzudt"
                  onClick={[Function]}
                  role="button"
                >
                  x
                </button>
              </CloseButtonstyles__Button>
            </CloseButton>
          </div>
        </Modalstyles__ModalWrapper>
        <Overlay
          className={null}
          onClick={[Function]}
        >
          <Overlaystyles__OverlayWrapper
            onClick={[Function]}
          >
            <div
              className="Overlaystyles__OverlayWrapper-sc-268q0n-0 giiUMR"
              onClick={[Function]}
            />
          </Overlaystyles__OverlayWrapper>
        </Overlay>
      </div>
    </Modalstyles__Wrapper>
  </Portal>
</Modal>
`;

exports[`Modal snapshots modal matches snapshot when custom styles are set 1`] = `
<Modal
  customClassNames={
    Object {
      "closeButton": "closeButtonClassName",
      "modal": "modalClassName",
      "overlay": "overlayClassName",
      "wrapper": "wrapperClassName",
    }
  }
>
  <Portal
    containerInfo={
      <body>
        <div
          class="Modalstyles__Wrapper-sc-5bf2l8-0 jIsDqI"
        >
          <div
            class="Modalstyles__ModalWrapper-sc-5bf2l8-1 ffLOAg"
          >
            <button
              class="CloseButtonstyles__Button-sc-1sd1qj0-0 pzudt"
              role="button"
            >
              x
            </button>
          </div>
          <div
            class="Overlaystyles__OverlayWrapper-sc-268q0n-0 giiUMR"
          />
        </div>
        <div
          class="Modalstyles__Wrapper-sc-5bf2l8-0 jIsDqI"
        >
          <div
            class="Modalstyles__ModalWrapper-sc-5bf2l8-1 caGlFr"
          >
            <button
              class="CloseButtonstyles__Button-sc-1sd1qj0-0 pzudt"
              role="button"
            >
              x
            </button>
          </div>
          <div
            class="Overlaystyles__OverlayWrapper-sc-268q0n-0 giiUMR"
          />
        </div>
        <div
          class="Modalstyles__Wrapper-sc-5bf2l8-0 jIsDqI"
        >
          <div
            class="Modalstyles__ModalWrapper-sc-5bf2l8-1 kVKwYK"
          >
            <button
              class="CloseButtonstyles__Button-sc-1sd1qj0-0 pzudt"
              role="button"
            >
              x
            </button>
          </div>
          <div
            class="Overlaystyles__OverlayWrapper-sc-268q0n-0 giiUMR"
          />
        </div>
        <div
          class="Modalstyles__Wrapper-sc-5bf2l8-0 jIsDqI wrapperClassName"
        >
          <div
            class="Modalstyles__ModalWrapper-sc-5bf2l8-1 ffLOAg modalClassName"
          >
            <button
              class="CloseButtonstyles__Button-sc-1sd1qj0-0 pzudt closeButtonClassName"
              role="button"
            >
              x
            </button>
          </div>
          <div
            class="Overlaystyles__OverlayWrapper-sc-268q0n-0 giiUMR overlayClassName"
          />
        </div>
      </body>
    }
  >
    <Modalstyles__Wrapper
      className="wrapperClassName"
    >
      <div
        className="Modalstyles__Wrapper-sc-5bf2l8-0 jIsDqI wrapperClassName"
      >
        <Modalstyles__ModalWrapper
          className="modalClassName"
          fluid={false}
          maxHeight={500}
          maxWidth={500}
          minHeight={100}
          minWidth={200}
        >
          <div
            className="Modalstyles__ModalWrapper-sc-5bf2l8-1 ffLOAg modalClassName"
          >
            <CloseButton
              className="closeButtonClassName"
              icon={null}
              onClick={[Function]}
            >
              <CloseButtonstyles__Button
                className="closeButtonClassName"
                onClick={[Function]}
                role="button"
              >
                <button
                  className="CloseButtonstyles__Button-sc-1sd1qj0-0 pzudt closeButtonClassName"
                  onClick={[Function]}
                  role="button"
                >
                  x
                </button>
              </CloseButtonstyles__Button>
            </CloseButton>
          </div>
        </Modalstyles__ModalWrapper>
        <Overlay
          className="overlayClassName"
          onClick={[Function]}
        >
          <Overlaystyles__OverlayWrapper
            className="overlayClassName"
            onClick={[Function]}
          >
            <div
              className="Overlaystyles__OverlayWrapper-sc-268q0n-0 giiUMR overlayClassName"
              onClick={[Function]}
            />
          </Overlaystyles__OverlayWrapper>
        </Overlay>
      </div>
    </Modalstyles__Wrapper>
  </Portal>
</Modal>
`;

exports[`Modal snapshots modal matches snapshot when maxHeight/minHeight/maxWidth/minWidth props are provided 1`] = `
<Modal
  maxHeight={100}
  maxWidth={120}
  minHeight={50}
  minWidth={60}
>
  <Portal
    containerInfo={
      <body>
        <div
          class="Modalstyles__Wrapper-sc-5bf2l8-0 jIsDqI"
        >
          <div
            class="Modalstyles__ModalWrapper-sc-5bf2l8-1 ffLOAg"
          >
            <button
              class="CloseButtonstyles__Button-sc-1sd1qj0-0 pzudt"
              role="button"
            >
              x
            </button>
          </div>
          <div
            class="Overlaystyles__OverlayWrapper-sc-268q0n-0 giiUMR"
          />
        </div>
        <div
          class="Modalstyles__Wrapper-sc-5bf2l8-0 jIsDqI"
        >
          <div
            class="Modalstyles__ModalWrapper-sc-5bf2l8-1 caGlFr"
          >
            <button
              class="CloseButtonstyles__Button-sc-1sd1qj0-0 pzudt"
              role="button"
            >
              x
            </button>
          </div>
          <div
            class="Overlaystyles__OverlayWrapper-sc-268q0n-0 giiUMR"
          />
        </div>
        <div
          class="Modalstyles__Wrapper-sc-5bf2l8-0 jIsDqI"
        >
          <div
            class="Modalstyles__ModalWrapper-sc-5bf2l8-1 kVKwYK"
          >
            <button
              class="CloseButtonstyles__Button-sc-1sd1qj0-0 pzudt"
              role="button"
            >
              x
            </button>
          </div>
          <div
            class="Overlaystyles__OverlayWrapper-sc-268q0n-0 giiUMR"
          />
        </div>
      </body>
    }
  >
    <Modalstyles__Wrapper>
      <div
        className="Modalstyles__Wrapper-sc-5bf2l8-0 jIsDqI"
      >
        <Modalstyles__ModalWrapper
          fluid={false}
          maxHeight={100}
          maxWidth={120}
          minHeight={50}
          minWidth={60}
        >
          <div
            className="Modalstyles__ModalWrapper-sc-5bf2l8-1 kVKwYK"
          >
            <CloseButton
              className={null}
              icon={null}
              onClick={[Function]}
            >
              <CloseButtonstyles__Button
                onClick={[Function]}
                role="button"
              >
                <button
                  className="CloseButtonstyles__Button-sc-1sd1qj0-0 pzudt"
                  onClick={[Function]}
                  role="button"
                >
                  x
                </button>
              </CloseButtonstyles__Button>
            </CloseButton>
          </div>
        </Modalstyles__ModalWrapper>
        <Overlay
          className={null}
          onClick={[Function]}
        >
          <Overlaystyles__OverlayWrapper
            onClick={[Function]}
          >
            <div
              className="Overlaystyles__OverlayWrapper-sc-268q0n-0 giiUMR"
              onClick={[Function]}
            />
          </Overlaystyles__OverlayWrapper>
        </Overlay>
      </div>
    </Modalstyles__Wrapper>
  </Portal>
</Modal>
`;
